Sources/NIOHTTP1/HTTPServerProtocolErrorHandler.swift

```diff
@@ -58,9 +58,11 @@ public final class HTTPServerProtocolErrorHandler: ChannelDuplexHandler, Removab
     public func write(context: ChannelHandlerContext, data: NIOAny, promise: EventLoopPromise<Void>?) {
         let res = self.unwrapOutboundIn(data)
         switch res {
-        case .head:
-            precondition(!self.hasUnterminatedResponse)
-            self.hasUnterminatedResponse = true
+        case .head(let head):
+            if head.status != .`continue` || self.hasUnterminatedResponse {
```

Comment:
This conditional seems wrong: if `hasUnterminatedResponse` is true we’ll enter this block and then perform two operations that are the exact duplicate of this operation. We previously preconditioned that `hasUnterminatedResponse` could not be set here. This is obviously no longer right with this patch, but the current state doesn’t seem better either.

I think the easiest choice is to remove the precondition and allow multiple HEADs. We could add tracking that verifies that we only really allow multiple 100s if we want to flesh out the feature.

Reply:
The conditional is proper. If a regular head has been issued, the precondition should fail even for 100's. It allows for any amount of 100's before, but not after, that.

`hasUnterminatedResponse` is only set for regular head's, that's what the condition is doing.

Reply:
Hmm, the conditional is very strange then. I might prefer the conditional to be constructed more broadly: it reads very strangely to have a conditional body that says “if x is true, precondition that x is false”. Perhaps a multi-step `if/else` would be clearer to read.

Reply:
OK, patched, the new version should be much clearer :-)